lib/ramble/ramble/spack_runner.py

```diff
@@ -602,16 +602,20 @@ def inventory_hash(self, require_exist=False):
         contents_to_hash = None
 
         if not self.dry_run:
-            if not lock_exists and (require_exist or len(self.env_contents) == 0):
+            if not lock_exists and require_exist and len(self.env_contents) > 0:
                 logger.die(
                     'spack.lock file does not exist in environment '
                     f'{self.env_path}\n'
                     'Make sure your workspace is fully setup with\n'
                     '    ramble workspace setup'
                 )
+            elif len(self.env_contents) == 0:
```

Comment:
Should this `elif` either:
1. Be nested under a `not lock_exists` test
or
2. Be below (but not nested under) the `elif lock_exists` test?

To make sure if a lock file exists, and no packages are defined we follow the logic for the existing lock file

Reply:
Looking at this again, I think this is actually not needed and this situation can be handled by:

```
617         if self.dry_run or not lock_exists:
```